sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala

```diff
@@ -0,0 +1,328 @@
+  }
+
+  def invalidTransformArgumentError(ctx: TransformArgumentContext): Throwable = {
+    new ParseException("Invalid transform argument", ctx)
+  }
+
+  val locationMsg = "please use the LOCATION clause to specify it"
+
+  val ownerMsg = "it will be set to the current user"
+
+  def cannotCleanReservedNamespaceProperty(
+      property: String, ctx: ParserRuleContext, msg: String): Throwable = {
+    new ParseException(s"$property is a reserved namespace property, $msg.", ctx)
+  }
+
+  def propertiesAndDbPropertiesBothSpecifiedError(ctx: CreateNamespaceContext): Throwable = {
+    new ParseException("Either PROPERTIES or DBPROPERTIES is allowed.", ctx)
+  }
+
+  def fromOrInNotAllowedInShowDatabasesError(ctx: ShowNamespacesContext): Throwable = {
+    new ParseException(s"FROM/IN operator is not allowed in SHOW DATABASES", ctx)
+  }
+
+  val providerMsg = "please use the USING clause to specify it"
+
+  def cannotCleanReservedTableProperty(
+      property: String, ctx: ParserRuleContext, msg: String): Throwable = {
+    new ParseException(s"$property is a reserved table property, $msg.", ctx)
+  }
+
+  def duplicatedTablePathsFoundError(
+      pathOne: String, pathTwo: String, ctx: ParserRuleContext): Throwable = {
+    new ParseException(s"Duplicated table paths found: '$pathOne' and '$pathTwo'. LOCATION" +
+      s" and the case insensitive key 'path' in OPTIONS are all used to indicate the custom" +
+      s" table path, you can only specify one of them.", ctx)
+  }
+
+  def storedAsAndStoredByBothSpecifiedError(ctx: CreateFileFormatContext): Throwable = {
+    new ParseException("Expected either STORED AS or STORED BY, not both", ctx)
+  }
+
+  def operationInHiveStyleCommandUnsupportedError(operation: String,
+      command: String, ctx: StatementContext, msgOpt: Option[String] = None): Throwable = {
+    val basicError = s"$operation is not supported in Hive-style $command"
+    val msg = if (msgOpt.isDefined) {
+      s"$basicError, ${msgOpt.get}."
+    } else {
+      basicError
+    }
+    new ParseException(msg, ctx)
+  }
+
+  val notNullOperation = "NOT NULL"
+
+  val alterColumnCommand = "ALTER COLUMN"
+
+  val replaceColumnsCommand = "REPLACE COLUMNS"
+
+  val alterTablePartitionReplaceColumnMsg =
+    "ALTER TABLE table PARTITION partition_spec REPLACE COLUMNS"
+
+  def operationNotAllowedError(message: String, ctx: ParserRuleContext): Throwable = {
+    new ParseException(s"Operation not allowed: $message", ctx)
+  }
+
+  def descColumnForPartitionUnsupportedError(ctx: DescribeRelationContext): Throwable = {
+    new ParseException("DESC TABLE COLUMN for a specific partition is not supported", ctx)
+  }
+
+  def incompletePartitionSpecificationError(
+      key: String, ctx: DescribeRelationContext): Throwable = {
+    new ParseException(s"PARTITION specification is incomplete: `$key`", ctx)
+  }
+
+  def isNotNoscanError(ctx: IdentifierContext): Throwable = {
```

Comment:
expectNoScanError

Reply:
How about `computeStatisticsNotExpectedError` ?

Reply:
Sounds good to me